src/spline.rs

```diff
@@ -42,6 +44,7 @@ impl<T, V> Spline<T, V> {
   {
     self
       .0
+      .make_contiguous()
```

Comment:
This is needed to make the deque a contiguous slice, necessary for sorting.
We should investigate whether it impacts performance considerably.